backend/tests/integration/common_utils/managers/user.py

```diff
@@ -0,0 +1,146 @@
+from copy import deepcopy
+from urllib.parse import urlencode
+from uuid import uuid4
+
+import requests
+
+from danswer.db.models import UserRole
+from danswer.server.manage.models import AllUsersResponse
+from danswer.server.models import FullUserSnapshot
+from danswer.server.models import InvitedUserSnapshot
+from ee.danswer.server.api_key.models import APIKeyArgs
+from tests.integration.common_utils.constants import API_SERVER_URL
+from tests.integration.common_utils.constants import GENERAL_HEADERS
+from tests.integration.common_utils.test_models import TestUser
+
+
+class UserManager:
+    @staticmethod
+    def create(
+        name: str | None = None,
+    ) -> TestUser:
+        if name is None:
+            name = f"test{str(uuid4())}"
+
+        email = f"{name}@test.com"
+        password = "test"
+
+        body = {
+            "email": email,
+            "username": email,
+            "password": password,
+        }
+        response = requests.post(
+            url=f"{API_SERVER_URL}/auth/register",
+            json=body,
+            headers=GENERAL_HEADERS,
+        )
+        response.raise_for_status()
+
+        test_user = TestUser(
+            id=response.json()["id"],
+            email=email,
+            password=password,
+            headers=deepcopy(GENERAL_HEADERS),
+        )
+        print(f"Created user {test_user.email}")
+
+        test_user.headers["Cookie"] = UserManager.login_as_user(test_user)
+
+        return test_user
+
+    @staticmethod
+    def login_as_user(test_user: TestUser) -> str:
+        data = urlencode(
+            {
+                "username": test_user.email,
+                "password": test_user.password,
+            }
+        )
+        headers = test_user.headers.copy()
+        headers["Content-Type"] = "application/x-www-form-urlencoded"
+
+        response = requests.post(
+            url=f"{API_SERVER_URL}/auth/login",
+            data=data,
+            headers=headers,
+        )
+        response.raise_for_status()
+        result_cookie = next(iter(response.cookies), None)
+
+        if not result_cookie:
+            raise Exception("Failed to login")
+
+        print(f"Logged in as {test_user.email}")
+        return f"{result_cookie.name}={result_cookie.value}"
+
+    @staticmethod
+    def add_api_key_to_user(
```

Comment:
I don't believe Admin keys aren't really associated with users. There should probably be a separate "type" of user called an "API Key" user (or an APIKeyManager)?

Reply:
done